### ISSUE TITLE
"Global volume" string is confusing

### ISSUE THREAD

**Issue by NABN00B** [hrydgard/ppsspp#18888]
The "Global volume" setting doesn't affect UI volume so it shouldn't be labeled as global, but rather "Game volume" or something.

On another note, the preview volume IS affected by the aforementioned setting, but I feel like this should be handled by a separate "UI volume" setting, which is something we don't have. See also #18865

**Comment by LunaMoo** [comment]
Not that it doesn't require changes, but a piece of history:
 this was called global volume, because it was needed for devices that doesn't have volume control and also, because in the past it had additional options for SAS and ATRAC/MP3 audio.
They were called differently I guess something alike "music" and "effects" don't remember exactly, but since they weren't doing what they were called(which was just a common use, but all games are different), people were crying about the setting not affecting what they wanted and as such the additional options got removed.

 It probably needs to be divided to:
- In-game volume with a sub-option of reverb volume (both prefferably as game specific settings),
- UI volume with a sub-option of achievements volume (through maybe those could just be normalized if the option exist purely because the achievements were too loud compared to other UI sounds) and possibly separate volume for preview songs.


 And maybe have a global volume on top of them all to keep it simple if there are still devices lacking volume control.

 Currently there's really no clear separation, just new things got new volume sliders without taking into account of what existed before and it got a bit messy.

**Comment by NABN00B** [comment]
I understand the history and I've come pretty much to the same conclusion as you LunaMoo.

In addition there could also be a tick for Advanced volume settings as unknown described here https://github.com/hrydgard/ppsspp/issues/7376#issuecomment-172445981

**Comment by cheako** [comment]
For devices like the steamdeck, where volume is controlled mainly by the OS.  The PSP should be locked so it always uses the greatest range, but a UI volume slider and mute(to replace enable/disable) would be nice.  So "game" volume should have a slider and a 3 position for max/adjusted/mute, perhaps just a slider with snap-to slots at the ends for these.

**Comment by hrydgard** [comment]
Yeah, this indeeds needs cleanup, along the lines of LunaMoo's proposal

**Comment by hrydgard** [comment]
I'm doing some changes here now: 

* Renaming "Global volume" to "Game volume", and making Alt Speed Volume a multiplier on top of that (so it's never louder).
* Adding UI volume which will control UI sounds (if enabled) and preview-music playback.

Also considering adding a lot more volume steps, 10 is quite coarse, even on a logarithmic scale. This will require renaming the ini keys of course (though can upgrade old settings).

However, I feel Achievement Volume should maybe also be a multiplier on top of Game Volume rather than part of UI sounds? While it's produced independently from the game, it does happen in response to an in-game event, so it feels like in-game audio.

**Comment by NABN00B** [comment]
A bit of late feedback. The improvements on the scaling are pretty good! 🚀 
However, when seeing it for the first time it's slightly confusing seeing 4 percentages stacked on top of each other. It isn't hinted which volume is dependent on which (or why Reverb volume has a different scale).

So if possible I'd recommend adding a line of text underneath each slider in the popup window where this would help. I.e. `Relative to / Multiplier of Game volume` when for Alt speed volume, an explanation for Reverb volume etc. (Also a minor thing but the sliders don't show the `%` metric like Button opacity or the others.)

For UI sounds, having the tickbox is redundant since setting UI volume to 0 mutes it.
I think we should have a separate option for the preview songs like LunaMoo has suggested. Although the loudness of them vary from game to game, being able to lower them in general (or mute them independently of UI SFX) would be an improvement.

As for Achievement volume, if you're gonna keep it independent of Game volume, then consider moving it to the UI sounds section.

Lastly, having Game volume default to 100 is a bit extreme. 📢 

I hope this helps.

**Comment by hrydgard** [comment]
The tickbox is actually not redundant since "UI Sounds" only affects the navigation sounds, not the "preview audio" of games (when selected with keyboard/gamepad or when in the info dialog). Not really sure how to solve that properly, but I guess maybe the checkbox should be changed into a "Navigation sounds" slider, separate from the preview audio? Not sure.

Well, games don't often actually reach 100% so it's not so extreme actually.

**Comment by NABN00B** [comment]
> Not really sure how to solve that properly, but I guess maybe the checkbox should be changed into a "Navigation sounds" slider, separate from the preview audio? Not sure.

Yes, that's what I implied.

**Comment by hrydgard** [comment]
I might do some more improvements for 1.19. At least the tricky part of rescaling the parameters is done. So, re-opening and renaming the issue.

**Comment by NABN00B** [comment]
"Alternate speed volume" has no default value, I don't think it needs a new issue so just throwing that in here.

**Comment by NABN00B** [comment]
"Achievement sound volume" and "UI sound" have their values applied before selecting OK in the popup.

**Comment by hrydgard** [comment]
Fixed those mostly, moving further work to 1.21.

**Comment by hrydgard** [comment]
Not really sure what's important that's left to do? I think we're not in a bad state right now.

**Comment by NABN00B** [comment]
> A bit of late feedback. The improvements on the scaling are pretty good! 🚀 However, when seeing it for the first time it's slightly confusing seeing 4 percentages stacked on top of each other. It isn't hinted which volume is dependent on which (or why Reverb volume has a different scale).
> 
> So if possible I'd recommend adding a line of text underneath each slider in the popup window where this would help. I.e. `Relative to / Multiplier of Game volume` when for Alt speed volume, an explanation for Reverb volume etc.
> 
> For UI sounds, having the tickbox is redundant since setting UI volume to 0 mutes it. 
> 
> As for Achievement volume, if you're gonna keep it independent of Game volume, then consider moving it to the UI sounds section.

Some of this is left.

And also
>"Achievement sound volume" and "UI sound" have their values applied before selecting OK in the popup.
